flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -1305,6 +1305,9 @@ class TaskManager(
         s"${task.getExecutionState} to JobManager for task ${task.getTaskInfo.getTaskName} " +
         s"(${task.getExecutionId})")
 
+      // delete all NAME_ADDRESSABLE BLOBs
+      libraryCacheManager.get.getBlobService.deleteAll(task.getJobID)
```

Comment:
Multiple tasks of the same job run in a TaskManager. This means that tasks delete each others blobs.